src/lib/mongoBackend/mongoRegistrationCreate.cpp

```diff
@@ -0,0 +1,219 @@
+* Author: Ken Zangelin
+*/
+#include <string>
+#include <vector>
+
+#include "mongo/client/dbclient.h"
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+#include "common/sem.h"
+#include "common/statistics.h"
+#include "common/errorMessages.h"
+#include "rest/OrionError.h"
+#include "rest/HttpStatusCode.h"
+#include "apiTypesV2/Registration.h"
+#include "mongoBackend/dbConstants.h"
+#include "mongoBackend/safeMongo.h"
+#include "mongoBackend/MongoGlobal.h"
+#include "mongoBackend/connectionOperations.h"
+#include "mongoBackend/mongoRegistrationCreate.h"
+
+
+
+/* ****************************************************************************
+*
+* setRegistrationId - 
+*/
+static void setRegistrationId(mongo::BSONObjBuilder* bobP, std::string* regIdP)
+{
+  mongo::OID oId;
+
+  oId.init();
+
+  *regIdP = oId.toString();
+  bobP->append("_id", oId);
+}
+
+
+
+/* ****************************************************************************
+*
+* setDescription -
+*/
+static void setDescription(const std::string& description, mongo::BSONObjBuilder* bobP)
+{
+  if (description != "")
+  {
+    bobP->append(REG_DESCRIPTION, description);
+  }
+}
+
+
+
+/* ****************************************************************************
+*
+* setExpiration -
+*/
+static void setExpiration(long long expires, mongo::BSONObjBuilder* bobP)
+{
+  if (expires != -1)
+  {
+    bobP->append(REG_EXPIRATION, expires);
+  }
+}
+
+
+
+/* ****************************************************************************
+*
+* setServicePath -
+*/
+static void setServicePath(const std::string& servicePath, mongo::BSONObjBuilder* bobP)
+{
+  if (servicePath != "")
+  {
+    bobP->append(REG_SERVICE_PATH, servicePath);
+  }
+}
+
+
+
+/* ****************************************************************************
+*
+* setContextRegistrationVector - 
+*/
+static void setContextRegistrationVector(ngsiv2::Registration* regP, mongo::BSONObjBuilder* bobP)
+{
+  mongo::BSONArrayBuilder  contextRegistration;
+  mongo::BSONArrayBuilder  entities;
+  mongo::BSONArrayBuilder  attrs;
+
+  for (unsigned int eIx = 0; eIx < regP->dataProvided.entities.size(); ++eIx)
+  {
+    ngsiv2::EntID* eP = &regP->dataProvided.entities[eIx];
+    std::string    eId;
+    std::string    eType;
+
+    eId           = (eP->idPattern   != "")? eP->idPattern : eP->id;
+    eType         = (eP->typePattern != "")? eP->typePattern : eP->type;
+
+    //
+    // FIXME PR: should patterns be supported or not?
+    //
+    // std::string  isIdPattern   = (eP->idPattern   != "")? "true" : "false";
+    // bool         isTypePattern = (eP->typePattern != "")? true : false;
+
+    if (eType == "")  // No type provided => all types
+    {
+      // entities.append(BSON(REG_ENTITY_ID << eId << REG_ENTITY_ISPATTERN << isIdPattern));
+      entities.append(BSON(REG_ENTITY_ID << eId));
+    }
+    else
+    {
+      // entities.append(BSON(REG_ENTITY_ID   << eId   << REG_ENTITY_ISPATTERN     << isIdPattern <<
```

Comment:
Leftover?

Reply:
All about patterns or no patterns, as the previous lines showed

Reply:
Fixed in 5686ca2